src/FluentValidation/AbstractValidator.cs

```diff
@@ -106,31 +103,27 @@ public abstract class AbstractValidator<T> : IValidator<T>, IEnumerable<IValidat
 		/// Note that cascade behaviour <i>within</i> individual rules is controlled by <see cref="RuleLevelCascadeMode"/>.
 		/// </para>
 		/// <para>
-		/// Do not set this to <see cref="CascadeMode.StopOnFirstFailure"/>.
-		/// <see cref="CascadeMode.StopOnFirstFailure"/> will be removed in a future release
-		/// and will throw an exception if used here.
+		/// This cannot be set to the deprecated <see cref="CascadeMode.StopOnFirstFailure"/>.
+		/// <see cref="CascadeMode.StopOnFirstFailure"/>. Attempting to do so it will actually
+		/// result in <see cref="CascadeMode.Stop"/> being used.
 		/// </para>
 		/// </summary>
 		public CascadeMode ClassLevelCascadeMode {
 			get => _classLevelCascadeMode();
-			set {
 #pragma warning disable 618
-				if (value == CascadeMode.StopOnFirstFailure) {
+			set => _classLevelCascadeMode = () => value == CascadeMode.StopOnFirstFailure

```

Comment:
@jsabrooke this is doing the conversion inside the delegate, rather than in the setter, meaning it'll happen on every invocation which isn't really needed. If it could be moved to the setter, then the comparison will only happen once.

Reply:
Yep, oops, thanks :-)